T/Torch/Torch/build_tarballs.jl

```diff
@@ -0,0 +1,271 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Torch"
+version = v"1.10.2"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/pytorch/pytorch.git", "71f889c7d265b9636b93ede9d651c0a9c4bee191"),
+    FileSource("https://micromamba.snakepit.net/api/micromamba/linux-64/0.21.1", "c907423887b43bec4e8b24f17471262c8087b7095683f41dcef4a4e24e9a3bbd"; filename = "micromamba.tar.bz2"),
+    ArchiveSource("https://github.com/JuliaBinaryWrappers/CUDA_full_jll.jl/releases/download/CUDA_full-v10.2.89%2B5/CUDA_full.v10.2.89.x86_64-linux-gnu.tar.gz", "60e6f614db3b66d955b7e6aa02406765e874ff475c69e2b4a04eb95ba65e4f3b"; unpack_target = "CUDA_full.v10.2"),
+    ArchiveSource("https://github.com/JuliaBinaryWrappers/CUDA_full_jll.jl/releases/download/CUDA_full-v11.3.1%2B0/CUDA_full.v11.3.1.x86_64-linux-gnu.tar.gz", "4094b8f1a3667166c1281faa7958cc46dbdc86ac86979e86d462a3c58f2a4b54"; unpack_target = "CUDA_full.v11.3"),
```

Comment:
Shouldn't these be regular dependencies? Maybe build only?

Reply:
They should be build dependecies, yes.

I believe there's an issue with specifying the version - I can't get one BuildDependency version for CUDA 10.2 and one for 11.3, I believe (I can't have two dependencies with the same name). It might be possible to just say `BuildDependency("CUDA_full_jll")` and get whatever version is most recent at the time (if the augmented platform implementation knows how to get CUDA_full 10.2 for platform with `cuda="10.2"`?).

I also figured out that the released libtorch binaries are built against a few more CUDA versions, so there might be some more platforms to add: https://github.com/pytorch/pytorch/runs/5467280928

Reply:
You're right you can't have together in the same environment, but you can limit a dependency to a specific set of platforms (which you're already doing for other dependencies)

Reply:
Giving it a shot in 6662c0877138ebe167f0154179b7dc043cd7c3f5

Reply:
Did not work:
```
ERROR: LoadError: it is invalid to specify multiple packages with the same name: `CUDA_full_jll [4f82f1eb]`
```
https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=21415&view=logs&j=f6c99a09-f697-57d4-7332-32215efd2007&t=239917b9-1794-551a-0432-a720bc1732c1&l=79

With deps:
```julia
    BuildDependency(PackageSpec("CUDA_full_jll", Base.UUID("4f82f1eb-248c-5f56-a42e-99106d144614"), v"10.2.89"); platforms = filter(p -> p["cuda"] == "10.2", cuda_platforms)),
    BuildDependency(PackageSpec("CUDA_full_jll", Base.UUID("4f82f1eb-248c-5f56-a42e-99106d144614"), v"11.3.1"); platforms = filter(p -> p["cuda"] == "11.3", cuda_platforms)),
```